src/app/api/v1/[...dialog]/route.ts

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+import { getComponentAsString } from '../../../functions/server/getComponentAsString'
+import AddEmailDialog from '../../../(nextjs_migration)/components/server/AddEmailDialog'
+
+export async function GET(req: NextRequest) {
+  const dialogUrl = req.url
+  const dialogType = dialogUrl.split('/').slice(-1)[0]
+  let dialogContentString = ''
+
+  switch (dialogType) {
+    case 'addEmail':
+      dialogContentString = await getComponentAsString({
+        component: AddEmailDialog()
+      });
+      break;
+    default:
+      throw new Error(`No dialog found for: ${dialogUrl}`);
```

Comment:
Q: who/where does this throw to? I'm assuming the Next routing system, but wasn't sure if this whole block should be in the `try..catch` block around L30-34 so we catch locally and use `NextResponse` w/ status 500.

Reply:
Yes, you are right. That’s a good point — done in [ab8c815](https://github.com/mozilla/blurts-server/pull/3142/commits/ab8c8152fbe1f6a82ab6ccb8e916a1bcecb24060).